File: Models/Artista.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace MSListasDeReproduccion.Models
{
    public partial class Artista
    {
        public Artista()
        {
            Albumes = new HashSet<Albume>();
        }

        public string Id { get; set; }
        public string Nombre { get; set; }
        public string NombreArtistico { get; set; }
        public int? AnoDeNacimiento { get; set; }
        public string Web { get; set; }
        public string Nacionalidad { get; set; }
        public uint FkIdEstatus { get; set; }

        public virtual Estatusderegistrosmusica FkIdEstatusNavigation { get; set; }
        public virtual ICollection<Albume> Albumes { get; set; }
    }
}

File: src/bd/entity/Artista.ts
import {Entity, PrimaryGeneratedColumn, Column,BaseEntity,OneToMany, PrimaryColumn} from "typeorm";
import {Album} from "./Album";
import {min,max,length,Length,Matches,IsNotEmpty,IsFQDN,IsNumber} from "class-validator";

@Entity("artistas")
export class Artista extends BaseEntity {
 
    
    @PrimaryColumn("varchar",{length:200})
    @IsNotEmpty()
    @Length(36,200)
    id: string;

    @Column("varchar",{length:200
                     ,comment:"nombre real del arista"
                      })
    @Matches("^([a-zA-zñ]+\\s{1,3})+[a-zA-Zñ]+\\s{0,3}$")
    nombre: string;

    @Column("varchar",{length:200
                      ,comment:"nombre artistico"
                      })
  
    @Matches("^([a-zA-z0-9]+\\s{0,3})+([a-zA-Z0-9]+\\s{0,3})*$")
    nombreArtistico: string;

    @Column("int",{nullable:true})
    anoDeNacimiento: number;

    @Column("longtext")
    @IsNotEmpty()
    @IsFQDN()
    web: string;
    
    @Column("varchar",{length:100})
    @Matches("^([a-zA-zñ]+|([a-zA-zñ]+\\s{1,3})+[a-zA-Zñ]+\\s{0,3})$")
    nacionalidad: string; 

    @Column("int",{ unsigned:true,
	 	            nullable:true,   
                    default:1
                       ,comment:"estado del registro (activo,inactivo)"
                   })
    @IsNumber()
    fkIdEstatus: number;

    

    @OneToMany(type => Album, album => album.artista,{cascade:true})
    albumes: Album[];
    
   
}

